ete4/smartview/renderer/gardening.pyx

```diff
@@ -16,7 +16,79 @@ def sort(tree, key=None, reverse=False):
     tree.children.sort(key=key, reverse=reverse)
 
 
-def root_at(node, bprops=None):
+def root_at(outgroup, bprops=None):
+    """ set the node "outgroup" as the first child of its current tree root """
+    import copy
+
+    t = outgroup.root
+    assert outgroup != t, 'cannot set the absolute tree root as an outgroup'
+
+    if len(t.children) > 2:
+        new_node = t.__class__()
+        for ch in list(t.children[1:]):
+            t.children.remove(ch)
+            new_node.children.append(ch)
+            ch.up = new_node
+
+        for bprop in ['support'] + (bprops or []):
+            if t.children[0].props.get(bprop, None) is not None:
+                new_node.props[bprop] = t.children[0].props[bprop]
+        t.children.append(new_node)
+        new_node.up = t
+
+
+    # if node is already a direct child of the root node, we  just make sure the
+    # outgroup is the "first" child and branch distances are balanced
+    if outgroup.up == t:
+        t.children.remove(outgroup)
+        t.children.insert(0, outgroup)
+
+        # balance distances
+        total_dist = (t.children[0].dist or 0.0) + (t.children[1].dist or 0.0)
+        t.children[0].dist = total_dist / 2.0
+        t.children[1].dist = total_dist / 2.0
+
+        return t
+
+    # if the outgroup is somewhere down in the tree structure, we want to rehang
+    # all branches but keep keep the original absolute tree root instance
+    # intact.
+    #
+    # So let's first detach current tree root from the tree and create another node
+    # instance (intermediate) that replaces it in the topology
+    intermediate_node = t.__class__()
+    intermediate_node.name = "inter"
+    for ch in t.children:
+        intermediate_node.add_child(ch)
+    t.children = []
+    t.up = None
+
```

Comment:
If I understand correctly, this is the only difference with the original code, which is that the reference to the old root (`t`) will still be a valid reference to the new root of the tree once it has been changed with the new outgroup.

If this is important (which is not obvious to me), then I think it may be easier to just swap the old root and the new root nodes at the very end.